.github/workflows/ci-pipeline.yml

```diff
@@ -20,15 +20,34 @@ on:
       - '.venv'
       - '.env'
       - '.vscode'
-      - '**/__pycache__'
       - '**/.pytest_cache'
+      - '**/__pycache__'
 jobs:
-  build:
-    name: Build and test
+  job-one:
+    name: build-and-run-test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: Build the "test" docker image
       run: docker build --target test --tag todo-app:test .
     - name: Run the "test" docker image
-      run: docker run --env-file ./.env.test todo-app:test
\ No newline at end of file
+      run: docker run --env-file ./.env.test todo-app:test
+  job-two:
+    name: deployment
+    needs: job-one
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_USERNAME : '${{ secrets.DOCKER_USERNAME }}'
+      DOCKER_PASSWORD : '${{ secrets.DOCKER_PASSWORD }}'
+      AZURE_WEBHOOK_URL_PASSWORD : '${{ secrets.AZURE_WEBHOOK_URL_PASSWORD }}'
+    if: ${{ github.event_name }} == 'push' && ${{ github.ref }} == 'refs/heads/exercise-9'

```

Comment:
Don't forget to update this before merging 😄 (Although you may find you're testing it out fairly regularly on future branches too!)